wallet/chain/p/builder/builder_test.go

```diff
@@ -0,0 +1,183 @@
+	require.Equal(expectedUnlockedUTXOs, utxosWithAssetID)
+	require.Equal(expectedLockedUTXOs, utxosWithOtherAssetID)
+}
+
+func TestSplitUTXOsByAssetID(t *testing.T) {
+	var (
+		require = require.New(t)
+
+		assetID                            = ids.GenerateTestID()
+		expectedUTXOsWithAssetIDPart0      = generateUTXOs(assetID, rand.Uint64())
+		expectedUTXOsWithAssetIDPart1      = generateUTXOs(assetID, rand.Uint64())
+		expectedUTXOsWithAssetIDPart2      = generateUTXOs(assetID, rand.Uint64())
+		expectedUTXOsWithOtherAssetIDPart0 = generateUTXOs(ids.GenerateTestID(), rand.Uint64())
+		expectedUTXOsWithOtherAssetIDPart1 = generateUTXOs(ids.GenerateTestID(), rand.Uint64())
+
+		utxos = utils.Join(
+			expectedUTXOsWithAssetIDPart0,
+			expectedUTXOsWithOtherAssetIDPart0,
+			expectedUTXOsWithAssetIDPart1,
+			expectedUTXOsWithOtherAssetIDPart1,
+			expectedUTXOsWithAssetIDPart2,
+		)
+		expectedUTXOsWithAssetID = utils.Join(
+			expectedUTXOsWithAssetIDPart0,
+			expectedUTXOsWithAssetIDPart1,
+			expectedUTXOsWithAssetIDPart2,
+		)
+		expectedUTXOsWithOtherAssetID = utils.Join(
+			expectedUTXOsWithOtherAssetIDPart0,
+			expectedUTXOsWithOtherAssetIDPart1,
+		)
+	)
+
+	utxosWithAssetID, utxosWithOtherAssetID := splitUTXOsByAssetID(utxos, assetID)
+	require.Equal(expectedUTXOsWithAssetID, utxosWithAssetID)
+	require.Equal(expectedUTXOsWithOtherAssetID, utxosWithOtherAssetID)
+}
+
+func TestUnwrapOutput(t *testing.T) {
+	normalOutput := &secp256k1fx.TransferOutput{
+		Amt: 123,
+		OutputOwners: secp256k1fx.OutputOwners{
+			Locktime:  456,
+			Threshold: 1,
+			Addrs:     []ids.ShortID{ids.ShortEmpty},
+		},
+	}
+
+	tests := []struct {
+		name             string
+		output           verify.State
+		expectedOutput   *secp256k1fx.TransferOutput
+		expectedLocktime uint64
+		expectedErr      error
+	}{
+		{
+			name:             "normal output",
+			output:           normalOutput,
+			expectedOutput:   normalOutput,
+			expectedLocktime: 0,
+			expectedErr:      nil,
+		},
+		{
+			name: "locked output",
+			output: &stakeable.LockOut{
+				Locktime:        789,
+				TransferableOut: normalOutput,
+			},
+			expectedOutput:   normalOutput,
+			expectedLocktime: 789,
+			expectedErr:      nil,
+		},
+		{
+			name: "locked output with no locktime",
+			output: &stakeable.LockOut{
+				Locktime:        0,
+				TransferableOut: normalOutput,
+			},
+			expectedOutput:   normalOutput,
+			expectedLocktime: 0,
+			expectedErr:      nil,
+		},
+		{
+			name:             "invalid output",
+			output:           nil,
+			expectedOutput:   nil,
+			expectedLocktime: 0,
+			expectedErr:      ErrUnknownOutputType,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			require := require.New(t)
+
+			output, locktime, err := unwrapOutput(test.output)
+			require.Equal(test.expectedOutput, output)
+			require.Equal(test.expectedLocktime, locktime)
+			require.ErrorIs(err, test.expectedErr)
```

Comment:
If there's an error it almost certainly guarantees that the other values are invalid, so check it first to avoid red herrings in test failures.

I also think the `output` and `lockTime` checks should use `assert` because simultaneous failure could provide more information, but not a hill I'll die on.